### ISSUE TITLE
Reproducibility of the results

### ISSUE THREAD

**Issue by wlaud1001** [Han-Jia/UNICORN-MAML#4]
Hi again, 

I'm interested in using pretrained weights in meta learning, and found your MAML perforamce is quite higher than others.

I tried to reproduce your results in my own code (Conv4, Miniimagenet, 5way 1shot), but the performance hardly improved (around 47~48) even with pretrained weight (from [here](https://drive.google.com/drive/folders/14Jn1t9JxH-CxjfWy4JmVpCxkC9cDqqfE)), increased number of adaptation steps, and higher learning rate.

So, I tried again with your code, but still got quite low performance even with the script you gave.

---
For ResNet, [miniimagenet](https://github.com/yaoyao-liu/mini-imagenet-tools), 5 way 1 shot,
```
python train_fsl.py --max_epoch 100 --way 5 --eval_way 5 --lr_scheduler step --model_class MAML --lr_mul 10 --backbone_class Res12 --dataset MiniImageNet --gpu 0 --query 15 --step_size 20 --gamma 0.1 --para_init './saves/initialization/miniimagenet/Res12-pre.pth' --lr 0.001 --shot 1 --eval_shot 1  --temperature 0.5 --gd_lr 0.05 --inner_iters 15
```
This script gave me this results (60.92), which is quite lower than your paper (64.42 in  your paper).

![image](https://user-images.githubusercontent.com/28920463/171568775-dcd2ef3c-e369-4a68-bc4a-cbb371d9e65a.png)

---
For [ConvNet](https://github.com/Han-Jia/AVIATOR/blob/main/model/networks/convnet_maml.py), [miniimagenet](https://github.com/yaoyao-liu/mini-imagenet-tools), 5 way 1 shot.
```
python train_fsl.py --max_epoch 100 --way 5 --eval_way 5 --lr_scheduler step --model_class MAML --lr_mul 10 --backbone_class ConvNet  --dataset MiniImageNet --gpu 9 --query 15 --step_size 20 --gamma 0.5 --para_init './saves/initialization/miniimagenet/con-pre.pth' --lr 0.002 --shot 1 --eval_shot 1 --temperature 0.5 --gd_lr 0.05 --inner_iters 15 
```
This script gave me this results (45.84), which is quite lower than your paper (54.89 in your paper).
![image](https://user-images.githubusercontent.com/28920463/171569875-f60c539e-7579-4429-96cf-b1d0e3f4a9fe.png)

---
If you don't mind, please check the reproducibility of your results, and could you please release the code with ConvNet network?

**Comment by wlaud1001** [comment]
Thank you for your reply!

I don't think it's a problem with the dataset I used.
I'll try again ConvNet with max-pooling layer.
Could you please share your ConvNet architecture code and script you used for?

**Comment by Han-Jia** [comment]
Hi,

The script of ConvNet is uploaded at [here](https://github.com/Han-Jia/UNICORN-MAML/blob/main/model/networks/convnet_maml.py).

**Comment by wlaud1001** [comment]
Thanks for your code.
But I still can't reproduce the performance.
I'll check later, thanks
![image](https://user-images.githubusercontent.com/28920463/173993438-e4930c6c-48d3-4e37-b6dc-580b393df403.png)

Best regards,
Jimyeong Kim
